engine/src/main/resources/assets/ui/menu/selectModsScreen.ui

```diff
@@ -33,6 +33,20 @@
                     }
                 },
                 {
+                    "type" : "UILabel",
+                    "id" : "warning",
+                    "text" : "WARNING: Enabling extra modules, especially all or those not in the stable lineup can cause game crashes or broken worlds.\n Please handle with care.",
```

Comment:
You can externalize the text to the [language pack files](https://github.com/MovingBlocks/Terasology/blob/develop/engine/src/main/resources/assets/i18n/menu_en.lang) and reference it e.g. as `${engine:menu#warn-modules}`.